kore/src/Kore/Simplify/Ceil.hs

```diff
@@ -361,3 +366,62 @@ makeSimplifiedCeil
 
         ~unexpectedError =
             error ("Unexpected term type: " ++ unparseToString termLike)
+
+{- | This returns a set of all subterms that need a Ceil
+
+When the term's ceil implies the ceils of its subterms, this also adds
+the subterms' simplified ceils to the result. This is needed because the
+SMT solver can't infer a subterm's ceil from a term's ceil, so we
+have to provide that information.
+
+As an example, if we call @enumerateSubtermsNeedingCeil@ for @f(g(x))@,
+ the return value will be @{ceil(f(g(x))), ceil(g(x)))}@.
+-}
```

Comment:
I think this description should be made a little clearer. we are not returning ceils, we are returning sets of all subterms which are assumed to be defined.